golang/cosmos/app/app.go

```diff
@@ -900,16 +901,9 @@ func (app *GaiaApp) initController(ctx sdk.Context, bootstrap bool) {
 	app.CheckControllerInited(false)
 	app.controllerInited = true
 
-	var blockTime int64 = 0
-	if bootstrap || app.upgradePlan != nil {
-		blockTime = ctx.BlockTime().Unix()
-	}
```

Comment:
I am uncomfortable with providing a `blockTime` and `blockHeight` for the init action sent during simple restart scenarios since there is so much non determinism in this case.

Reply:
There is no bug. It's just how we use the init "blocking send", and that a node can be started either after consensus stop, or from per validator action. I was simply concerned about including consensus state in actions happening outside of consensus. But you're right, there isn't really a problem as long as no consensus decision is taken from that data.

Reply:
Thanks.  I will add a comment describing that to the init blocking send call.